### ISSUE TITLE
Changing a strategy parameter in Deployment from Recreate to RollingUpdate 

### ISSUE THREAD

**Issue by ysakashita** [NetApp/trident#435]
**Describe the solution you'd like**

I want to change a strategy parameter of trident-csi deployment, which is created by Trident Operator, from `Recreate` to `RollingUpdate`.
When a trident-csi pod from deployments is updating, trident-csi from daemonset occor this error.

```
$ kubectl rollout restart deployment trident-csi 
$ starn trident-csi
...
trident-csi-qtfjb trident-main time="2020-08-20T02:53:31Z" level=debug msg="Controller not yet ready, waiting." error="could not log into the Trident CSI Controller: error communicating with Trident CSI Controller; Put \"https://10.27.2.48:34571/trident/v1/node/demo-netapptest-master-default-e5534874-khqsw\": dial tcp 10.27.2.48:34571: connect: connection refused" increment=989.409747ms
...
```

I think that this is correct behavior.
But I would like to reduce connection loss during trident-csi update.

**Describe alternatives you've considered**

In other idea, you try to change number of replicas.

**Additional context**

none

**Comment by gnarl** [comment]
@ysakashita,

We've looked at this and want to know more about the use case. We do not understand the advantage of using RollingUpdate here when the Trident Controller is not yet ready during a Trident upgrade. Can you provide more information.

**Comment by ysakashita** [comment]
@gnarl We want to avoid trident-csi downtime as much as possible.
As you may know, the current strategy(`Recreate`) in Deployment work as follows.

https://kubernetes.io/docs/concepts/workloads/controllers/deployment/

```
All existing Pods are killed before new ones are created when .spec.strategy.type==Recreate.
```

Therefore, downtime for trident-csi is longer.

On the other hand, the strategy(`RolliongUpdate`) is able to control  the rolling update process. (e.g. maxUnavailable)
This will reduce downtime because the old trident-csi will not be lost until the new trident-csi is ready.

**Comment by gnarl** [comment]
Hi @ysakashita,

I've talked to the team about this issue. Since Trident is a stateful application the running Trident deployment must be stopped before the upgraded Trident deployment can be started. That said we are looking into the performance of the Trident daemonset registering with the Trident controller. While this doesn't address the logged error you reported we may be able to improve the overall upgrade performance for a K8S cluster with a large number of worker nodes.

**Comment by ysakashita** [comment]
I am fine with any method as long as it reduces the downtime of trident-csi. 👍

**Comment by gnarl** [comment]
This issue is fixed with [commit](https://github.com/NetApp/trident/commit/b3d6165b29dc6d9806ccfc2a3c4c4fc9dee9d103) and will be included in the Trident 21.07 release.
